### PR TITLE
Add env vars for mocked service urls

### PR DESCRIPTION
**Pull request by david-martin**
This change does a few things:

* Brings in the openshiftResourceParser library as a component instead
of an external dependency
* Modifies the parser to use a hardcoded list of services available i.e.
Fuse, Che, Launcher, EnMasse
* Sets the link for each of these services according to 4 new *build time* env vars (not runtime env vars):
  * REACT_APP_FUSE_URL
  * REACT_APP_LAUNCHER_URL
  * REACT_APP_CHE_URL
  * REACT_APP_ENMASSE_URL

A deploy of this can be see http://react-demo-app-webapp.cluster-lfa3xlh.opentry.me/ (This still uses the s2i build method with nginx for serving content)
The BuildConfig has these env vars set to the corresponding urls for those services in that cluster.

**NOTE** This change is not intended for production use. It is only for using configurable urls with a fixed mocked list of 4 services. Auto-discovery of services & their console urls will be added in a separate PR.

### CODE REVIEW

**Comment by david-martin** [comment]
@johnfriz @aidenkeating would you mind taking a look (url & details in description)

**Comment by priley86** [comment]
this is looking good to me... will test some locally here shortly.

File: src/components/openshiftResourceParser/clients/mock-provisioned-service-client.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        appDescription: 'A developer workspace server and cloud IDE.',
+        appLink: `${process.env.REACT_APP_CHE_URL}`
+      },
+      {
+        appName: 'EnMasse',
+        appDescription: 'Managed, self-service messaging on Kubernetes.',
+        appLink: `${process.env.REACT_APP_ENMASSE_URL}`
+      }
+    ];
+    debugger
```
Comment (dlabaj) [inline src/components/openshiftResourceParser/clients/mock-provisioned-service-client.js]: remove debugger

File: src/pages/landing/landingPage.js [hunk 2]
```diff
@@ -2,53 +2,36 @@ import * as React from 'react';
...
-  </div>
-);
+class LandingPage extends React.Component {
+  state = {
+    apps: []
+  }
+
+  componentDidMount() {
+    const parser = new OpenShiftResourceParser({ mockData: true });
+    parser.listProvisionedMWServices("eval")
```
Comment (priley86) [inline src/pages/landing/landingPage.js]: per discussion w/ @dlabaj, would you be opposed to us porting `listProvisionedMWServices` into a Redux action in a future PR? The `OpenShiftResourceParser` and relevant services should be fine as is, we'd just like to hook into this `listProvisionedMWServices ` promise from a Redux action/reducer (and treat this like an API service). For now, this should work fine though! Redux is _optional_.

File: src/components/openshiftResourceParser/clients/mock-provisioned-service-client.js (hunk 1 above)
Comment (david-martin) [reply src/components/openshiftResourceParser/clients/mock-provisioned-service-client.js]: :+1:

File: src/pages/landing/landingPage.js (hunk 2 above)
Comment (david-martin) [reply src/pages/landing/landingPage.js]: No objections. (I'll have to do some reading on redux to fully grok what you are saying :) )

**Comment by priley86** [comment]
@david-martin ok to merge? I think we are OK when you are.

File: src/pages/landing/landingPage.js (hunk 2 above)
Comment (priley86) [reply src/pages/landing/landingPage.js]: I think Redux definitely makes sense if we are also looking to use this list to drive the application launcher menu. This is a case where Redux's global state is perfect. Will follow up in future PR.

File: package.json [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
     "axios": "^0.18.0",
     "classnames": "^2.2.5",
+    "client-oauth2": "^4.2.1",
     "detect-browser": "^3.0.0",
     "express": "^4.16.3",
     "i18next": "^11.6.0",
     "i18next-xhr-backend": "^1.5.1",
     "js-cookie": "^2.2.0",
     "lodash.get": "^4.4.2",
-    "openshift-resource-parser": "^0.1.0",
```
[the remark this answers is not in the document]
Comment (david-martin) [reply package.json]: Can discuss further in https://github.com/integr8ly/openshift-resource-parser/issues/2 if you wish.
I wouldn't like to have it as a separate lib if it is going to become heavily tied to the tutorial web app (unnecessary churn of module build/release cycle).
Perhaps there's a piece of it (just the openshift api & auth stuff?) that could be pulled out, but I'd like to see that naturally happen over time, if that makes sense

**Comment by david-martin** [comment]
Thanks @dlabaj  @priley86.
I'd like to create a PR that swaps out nginx for the node.js server (and BrowserRouter instead of HashRouter) next, which will include CI build for a docker image.

We can hold off on the auth/sso PR #7 a little longer as there's env config needed for it as part of that work.

**Pull request merged**